apps/dashboard/app/apps/request_tracker_client.rb

```diff
@@ -0,0 +1,128 @@
+# frozen_string_literal: true
+
+require 'rest_client'
+
+# HTTP client to create a request tracker ticket using the API
+# Configuration parameters:
+# - `server`: URL for the Request Tracker server (required)
+# - `user`: RT API username
+# - `pass`: RT API password
+# - `auth_token`: RT API [auth token](https://github.com/bestpractical/rt-authen-token), preferred instead of username and password.
+# - `timeout`: Connection and read timeout in seconds. Defaults to 30.
+# - `verify_ssl`: Whether or not the client should validate SSL certificates. Defaults to true.
+# - `proxy`: Proxy server URL. Defaults to no proxy.
+#
+class RequestTrackerClient
+  # THIS CLIENT IS BASED ON https://github.com/uidzip/rt-client
+
+  def self.create
+    rt_config = ::Configuration.support_ticket_config.fetch(:rt_api, {})
+
+    new({
+          server:     rt_config[:server],
+          user:       rt_config[:user],
+          pass:       rt_config[:pass],
+          auth_token: rt_config[:auth_token],
+          timeout:    rt_config[:timeout],
+          verify_ssl: rt_config[:verify_ssl],
+          proxy:      rt_config[:proxy]
+        })
+  end
+
+  UA = 'Open OnDemand ruby RT Client'
+  attr_reader :server, :rt_client, :resource, :timeout, :verify_ssl
+
+  def initialize(config)
+    # DEFAULTS
+    @auth_token = nil
+    @timeout = 30
+    @verify_ssl = false
+    # FROM CONFIGURATION
+    @user = config[:user]
+    @pass = config[:pass]
+    @auth_token = config[:auth_token]
+    @timeout = config[:timeout] if config[:timeout]
```

Comment:
Can you do something like this if there are defaults?

```suggestion
    @timeout = config[:timeout] || 30
```

Reply:
Updated defaults setup